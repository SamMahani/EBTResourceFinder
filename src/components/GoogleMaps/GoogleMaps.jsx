import React from "react";
import { Map, InfoWindow, Marker, GoogleApiWrapper } from "google-maps-react";
import GMA from "../../../GMA.json";
export class MapContainer extends React.Component {
  render() {
    let points = this.props.list.reduce((acc, store) => {
      let lat = Number(store.latitude);
      let lng = Number(store.longitude);
      let coordinate = {
        lat,
        lng
      };
      acc.push(coordinate);
      return acc;
    }, []);
    var bounds = new this.props.google.maps.LatLngBounds();
    for (var i = 0; i < points.length; i++) {
      bounds.extend(points[i]);
    }
    return (
      <Map
        google={this.props.google}
        center={{
          lat: this.props.currLat,
          lng: this.props.currLong
        }}
        bounds={bounds}
      >
        {this.props.list.map(store => {
          return (
            <Marker
              title={store.store_name}
              name={store.store_name}
              position={{ lat: store.latitude, lng: store.longitude }}
              key={`${store.address}_${store.latitude}_${
                store.longitude
              }${Math.random()}`}
            />
          );
        })}

        {/* <InfoWindow onClose={this.onInfoWindowClose}>
          <div>
            <h1 />
          </div>
        </InfoWindow> */}
      </Map>
    );
  }
}

export default GoogleApiWrapper({
  apiKey: GMA.GMA
})(MapContainer);
