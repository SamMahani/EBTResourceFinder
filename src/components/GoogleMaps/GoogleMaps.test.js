import React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";
import GoogleMaps from "./GoogleMaps.jsx";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<GoogleMaps />);
  expect(asFragment()).toMatchSnapshot();
});
