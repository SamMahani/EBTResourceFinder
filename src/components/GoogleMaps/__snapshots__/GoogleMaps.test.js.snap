// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
